src/trace.ml

```diff
@@ -272,23 +272,46 @@ module Make_commands (Backend : Backend_intf.S) = struct
   let record_dir_flag mode =
     let open Command.Param in
     flag
-      "-record-dir"
+      "-temporary-data-directory"
       (mode Filename_unix.arg_type)
-      ~doc:"DIR create this directory if necessary and put raw trace data in it"
+      ~doc:
+        "DIR Where to store temporary files (including raw perf.data files). If not \
+         provided, magic-trace stores temporary files in a subdirectory of /tmp and \
+         deletes them when it's done. If provided, files will be stored in the given \
+         directory, creating the directory if necessary, and magic-trace will not delete \
+         the directory when it's done."
   ;;
 
+  (*
+      let record_dir, cleanup =
+        match record_dir with
+        | Some dir ->
+          if not (Sys_unix.is_directory_exn dir) then Core_unix.mkdir dir;
+          dir, false
+        | None -> Filename_unix.temp_dir "magic_trace" "", true
+      in
+  *)
+
   let record_flags =
     let%map_open.Command record_dir = record_dir_flag optional
     and executable_override =
       flag
-        "-executable-override"
+        "-override-executable"
```

Comment:
I don't feel strongly about this, but I don't know where this flag would be useful.